### PR TITLE
Add train_image_classifier.py for training image classifier

### PR DESCRIPTION
**Pull request by Sonnyburni**

### CODE REVIEW

File: model_development/train_image_classifier.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+        ]
+    ),
+}
+
+
+def get_dataloaders(data_dir, batch_size=32):
+    dataset = datasets.ImageFolder(data_dir, data_transforms["train"])
```
Comment (filippovarini) [inline model_development/train_image_classifier.py]: Is it intended to apply the train transformation to all data? (train and val)?

File: model_development_pipeline.py [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
 # Construct Image Classification Dataset
 construct_image_classification_dataset(object_detection_yaml)
 
 # %%
 # Aggregate the object_detection classes in a single "elasmobranch" class
 aggregate_all_classes(object_detection_yaml)
 
+# %%
+# Path to the image classification folder
+image_classification_folder = object_detection_yaml.parent / "image_classification"
```
Comment (filippovarini) [inline model_development_pipeline.py]: If there is already an "./image_classification" folder, then the `construct_image_classification_dataset` will create a new `./image_classification1` folder. 
Therefore, you can't assume the folder will always be "./image_classification". 

Can you make  construct_image_classification_dataset() return the image_classification path and use that?

File: model_development/train_image_classifier.py (hunk 1 above)
Comment (Sonnyburni) [reply model_development/train_image_classifier.py]: Thanks for flagging @filippovarini  - updated to apply correct transforms to both train and val now

File: model_development_pipeline.py (hunk 2 above)
Comment (Sonnyburni) [reply model_development_pipeline.py]: @filippovarini updated this now :)

File: model_development/train_image_classifier.py [hunk 3]
```diff
@@ -45,11 +45,20 @@ def get_dataloaders(data_dir, batch_size=32):
         train_subset = Subset(dataset, train_ids)
         val_subset = Subset(dataset, val_ids)
         train_loader = DataLoader(
-            train_subset, batch_size=batch_size, shuffle=True, num_workers=4
+            train_subset,
+            batch_size=batch_size,
+            shuffle=True,
+            num_workers=4,
+            transform=data_transforms["train"],
```
Comment (filippovarini) [inline model_development/train_image_classifier.py]: Can you pass transformations in DataLoader? They seem to not be a defined parameters [here](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader).

In any case, both dataloaders are from subsets of the same dataset, which you define with dataset.ImageFolder(..., data_transforms["train"]). So both have the train transformations.

File: model_development_pipeline.py [hunk 4]
```diff
@@ -1,26 +1,32 @@
...
 workspace = "msc-fish-seychelles"
 project = "seychelles"
 version = 35
 object_detection_yaml = download_dataset(workspace, project, version, "yolov8")
 
-#%%
+# %%
 # Construct Image Classification Dataset
-construct_image_classification_dataset(object_detection_yaml)
+image_classification_folder = construct_image_classification_dataset(
```
Comment (filippovarini) [inline model_development_pipeline.py:16]: I think currently construct_image_classification_dataset does not return the folder. Could you update the code to make it return?

File: model_development/train_image_classifier.py (hunk 3 above)
Comment (filippovarini) [reply model_development/train_image_classifier.py]: @Sonnyburni

File: model_development_pipeline.py (hunk 4 above)
Comment (filippovarini) [reply model_development_pipeline.py:16]: @Sonnyburni

**Review by filippovarini — APPROVED** [review]

**Review by filippovarini — APPROVED** [review]

**Pull request merged**